### PR TITLE
MGDAPI-5811 fix the csv locator function

### PR DESCRIPTION
**Pull request by MStokluska**
# Issue link
https://issues.redhat.com/browse/MGDAPI-5811

# What
The subscription controller had an issue where the CSV locator function was failing to fetch CSV from cm binaryData.

# Verification steps
- Provision 2 osd clusters
- On both clusters create catalog source pointing to quay.io/mstoklus/managed-api-service:v1.36.0 - matching current prod
- Install RHOAM

## On cluster 1
- once installation is completed, confirm that the subscription for RHOAM has approvals set to "Automatic"
```
oc get sub managed-api-service -n redhat-rhoam-operator -o yaml 
```
- update index to point to quay.io/mstoklus/managed-api-service:v1.37.0
- confirm upgrade has been completed successfully and that the subscription approval is updated to "Manual"
```
oc get sub managed-api-service -n redhat-rhoam-operator -o yaml
``` 
## On cluster 2
- once installation is completed update the approval in subscription configuration to "Manual"
```
oc edit sub managed-api-service -n redhat-rhoam-operator -o yaml 
```
- once updated, update the catalog source image to quay.io/mstoklus/managed-api-service:v1.37.0
- confirm the upgrade has completed successfully after manually approving 1.37.0

### CODE REVIEW

File: controllers/subscription/csvlocator/csvlocator.go [hunk 1]
```diff
@@ -78,59 +78,45 @@ func (l *ConfigMapCSVLocator) GetCSV(ctx context.Context, client k8sclient.Clien
...
+				return nil, fmt.Errorf("failed to decode base64: %s", err)
+			}
+
+			// Decompress from gzip
+			reader, err := gzip.NewReader(bytes.NewBuffer(compressedData))
+			if err != nil {
+				return nil, err
+			}
+
+			result, err := ioutil.ReadAll(reader)
```
Comment (adam-cattermole) [inline controllers/subscription/csvlocator/csvlocator.go:94]: nit:

> Deprecated: As of Go 1.16, this function simply calls io.ReadAll.


```suggestion
			result, err := io.ReadAll(reader)
```

**Comment by adam-cattermole** [comment]
Verified both scenarios:

* Cluster with automatic sub immediately upgraded by OLM and subscription switched to manual
* Cluster with manual sub required manual approval
* Upgrades for both complete successfully

Checked logs in both clusters pre and post upgrade and can no longer see error in subscription controller post upgrade. Added minor comment on deprecated function but happy for it to be addressed at a future date.

/lgtm

**Comment by MStokluska** [comment]
@adam-cattermole thanks for the review, I've updated the ioutil - good catch.
Please add lgtm again if you can.

**Comment by adam-cattermole** [comment]
Thanks @MStokluska 

/lgtm

**Pull request merged**